src/sage/sets/set.py

```diff
@@ -1085,6 +1082,32 @@ def __richcmp__(self, other, op):
             return rich_to_bool(op, 0)
         return rich_to_bool(op, -1)
 
+    def isdisjoint(self, other):
+        """
+        Return whether ``self`` and ``other`` are disjoint.
+
+        INPUT:
+
+        - ``other`` -- a finite Set
+
+        EXAMPLES::
+
+            sage: X = Set([1,2,3])
+            sage: Y = Set([2,4,6])
+            sage: Z = Set([4,5,6])
+            sage: X.isdisjoint(Y)
+            False
+            sage: X.isdisjoint(Z)
+            True
+            sage: Y.isdisjoint(Z)
+            False
+        """
+        if not isinstance(other, Set_object_enumerated):
+            raise NotImplementedError
+        return self.set().isdisjoint(other.set())
```

Comment:
I think we should be much less restrictive here.  For example,

```
sage: s = set([1,2,3])
sage: s.is_disjoint(range(4,6))
False
```
just works, and I see no good reason why it shouldn't.  So possibly simply (untested!):
```suggestion
        return self.set().isdisjoint(other)
```

Reply:
I would like to understand what happens eg when other is not an enumerable Set, eg an infinite Set such as `Set(QQ)`? Should we put a non-finite check instead?

Further, this pattern is quite similar to the existing below pattern -
```
    def issubset(self, other):
        if not isinstance(other, Set_object_enumerated):
            raise NotImplementedError
        return self.set().issubset(other.set())
```

Then, just to be consistent, should we update these ones as well?

Reply:
Certainly `Set([1,2,3]).isdisjoint(ZZ)` works and returns `False` as per your suggestion. However, `Set([1,2,3]).isdisjoint(RR)` fails with `NotImplementedError`. So I guess it seems okay to modify others as well?